doc-surrealdb_versioned_docs/version-2.x/reference-guide/security-best-practices.mdx

```diff
@@ -0,0 +1,323 @@
+---
+sidebar_position: 4
+sidebar_label: Security Best Practices
+title: Security Best Practices | Reference guides
+description: This guide outlines some key security best practices for using SurrealDB 2.0. While SurrealDB offers powerful and flexible features to support you in meeting your desired security standards, the use that you make of those features will ultimately determine whether or not you meet them.
+---
+import ThemedImage from '@theme/ThemedImage';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import { SurrealistMini } from "@site/src/components/SurrealistMini";
+
+# Security Best Practices
+
+This guide outlines some key security best practices for using SurrealDB 2.0. While SurrealDB offers powerful and flexible features to support you in meeting your desired security standards, the use that you make of those features will ultimately determine whether or not you meet them. The following is a non-exhaustive list of security best practices that you should take into account when building services and applications with SurrealDB to help you address common security challenges while preventing frequent pitfalls.
+
+## Capabilities
+
+When running a SurrealDB server, you are able to configure the [capabilities](https://surrealdb.com/docs/surrealdb/security/capabilities) that it will have. These capabilities are disabled by default to expose as little attack surface as possible to malicious actors. We strongly discourage running SurrealDB with all capabilities enabled without a good reason for it. Instead, we recommend enabling only the capabilities that are necessary to your service.
+
+Whenever you need to enable a specific capability, we recommend that you do it on a case by case basis instead of generally. For example, if you know that your queries need to be able to parse emails using functions, we recommend you run SurrealDB with the `--allow-funcs parse::email::*`  flag instead of allowing all functions with `--allow-funcs` without any arguments. This ensures that your SurrealDB instance will not be affected by vulnerabilities in the code of any other functions that a malicious actor could leverage to increase their privileges or execute unauthorized code.
+
+**Example**
+
+```bash
+# Allow SurrealDB to generate UUIDv4 and call any crypto function except MD5.
+surreal start --allow-funcs "crypto","rand::uuid::v4" --deny-funcs "crypto::md5"
+```
+
+Such a practice becomes specially important in the case of the network capability, which allows SurrealDB to perform network requests such as those required by the [`http::*` functions](https://surrealdb.com/docs/surrealdb/2.x/surrealql/functions/database/http). Allowing untrusted users to perform network requests from your SurrealDB instance can allow them access to its local network or services that specifically allow network access from that instance.
```

Comment:
```suggestion
Using capabilities becomes especially important in the case of the network capability, which allows SurrealDB to perform network requests such as those required by the [`http::*` functions](/docs/surrealdb/2.x/surrealql/functions/database/http). Allowing untrusted users to perform network requests from your SurrealDB instance can allow them access to its local network or services that specifically allow network access from that instance.
```

Reply:
By "such a practice" I meant only allowing specific instances of a capability.